### ISSUE TITLE
[no-unsafe-return] option to allow returning any if the function explicitly declares returning any

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#3482]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unsafe-return": "error"
  }
}
```

```TS
function returnsAnyIntentionally(input: string): any {
  return JSON.parse(input);
}
```

Practically, see `removeLocationDataAndSourceTypeFromProgramNode` internally. https://github.com/typescript-eslint/typescript-eslint/blob/9524424a4bce09dafb7eb0d6c322f07f2a0003da/packages/typescript-estree/tests/ast-alignment/utils.ts

**Expected Result**

Suppose you have a function that intentionally has a `: any` return type declared and returns something that is `any`. 

Can we add an option to the rule like `allowExplicitAny` that prevents a rule complaint if the function has that `: any`?

**Actual Result**

This rule still gives a complaint on that, even though the `any` is _intentional_.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.26.0` |
| `@typescript-eslint/parser`        | `4.26.0` |
| `TypeScript`                       | `4.3.2` |
| `ESLint`                           | `7.27.0` |
| `node`                             | `12.16.0` |

**Comment by bradzacher** [comment]
this should work this way!

you can do it for generic arguments, so I'm surprised you can't do it for straight `any`.
https://github.com/typescript-eslint/typescript-eslint/blob/fc5f171b1ade2b45a1a4268b6d22926d420282a6/packages/eslint-plugin/tests/rules/no-unsafe-return.test.ts#L46-L50

**Comment by HolgerJeromin** [comment]
Is this expected? Or should I open another Item?
```ts
function foo2(): any {
    if (Math.random()) {
        let res: any = {};
        return res;
    }
    let resArr: any[] = [];
    resArr.push(true);
    return resArr; // <-- eslint unhappy
}

**Comment by JoshuaKGoldberg** [comment]
Hi @HolgerJeromin! In general I'd suggest filing another issue, as this one has already been closed.
